scripts/us_census/pep/us_pep_sex/process.py

```diff
@@ -1103,54 +1107,54 @@ def add_future_year_urls():
-                                _FILES_TO_DOWNLOAD.append(
-                                    {"download_path": url_to_check})
-
-                        except requests.exceptions.RequestException as e:
-                            logging.fatal(
-                                f"URL is not accessible {url_to_check} due to {e}"
-                            )
-
-                else:  # This URL does not contain {i}, so we only need to process it once per year
-                    url_to_check = url.format(YEAR=YEAR)
-
-                    # If the URL has already been processed for this year, skip it
-                    if url_to_check in downloaded_year_urls:
-                        continue  # Skip this URL if it's already processed
+    # Loop through years in reverse order from 2030 to 2023
+    for future_year in range(2030, 2022, -1):  # From 2030 to 2023
+
+        YEAR = future_year
+        # Loop through URLs
+        for url in urls_to_scan:
+            if "{i}" in url:  # This URL contains the {i} variable, so we loop through i from 01 to 56
+                for i in range(1, 57):  # Loop i from 01 to 56
+                    formatted_i = f"{i:02}"  # Ensure i is always 2 digits (01, 02, ..., 56)
+                    url_to_check = url.format(YEAR=YEAR, i=formatted_i)
+                    logging.info(f"checking url: {url_to_check}")
 
                     try:
                         check_url = requests.head(url_to_check,
                                                   allow_redirects=True)
                         if check_url.status_code == 200:
                             _FILES_TO_DOWNLOAD.append(
                                 {"download_path": url_to_check})
-                            downloaded_year_urls.add(
-                                url_to_check)  # Mark this URL as processed
-
-                        else:
-                            logging.fatal(
-                                f"URL returned status code {check_url.status_code}: {url_to_check}"
-                            )
 
                     except requests.exceptions.RequestException as e:
-                        logging.fatal(
+                        logging.error(
                             f"URL is not accessible {url_to_check} due to {e}")
 
+            else:  # This URL does not contain {i}, so we only need to process it once per year
+                url_to_check = url.format(YEAR=YEAR)
+                logging.info(f"checking url: {url_to_check}")
+                # If the URL has already been processed for this year, skip it
+                if url_to_check in downloaded_year_urls:
+                    continue  # Skip this URL if it's already processed
+
+                try:
+                    check_url = requests.head(url_to_check,
+                                              allow_redirects=True)
+                    if check_url.status_code == 200:
+                        _FILES_TO_DOWNLOAD.append(
```

Comment:
If we find a valid set of URLs for a later year, say 2023, shouldn't we stop adding more _FILES_TO_DOWNLOAD such as for 2022, 2021, etc?

Reply:
Resolving to kick off an import run. Pls fix in a follow PR if needed.